### PR TITLE
Clean up + document savedata/save_table.c

### PR DESCRIPTION
**Pull request by lhearachel**
## Files

* `include/unk_0202440C.h` -> `include/savedata/save_table.h`
* `src/unk_0202440C.c` -> `src/savedata/save_table.c`

## Structs

* `UnkStruct_021C0794` -> `SaveData`
* `UnkStruct_0202440C` -> `TVBroadcast`
* `UnkStruct_020797DC` -> `PSSBoxes`
* `UnkStruct_0202442C` -> `MysteryGift`
* `UnkStruct_02024440` -> `PalParkTransfer`
* `UnkStruct_0202DF8C` -> `HallOfFame`
* `UnkStruct_0202F264` -> `BattleRecording`
* `UnkStruct_020308A0` -> `BattleFrontierStage`
* `UnkStruct_02025340` -> `SaveDataTableEntry`

## Functions

* `sub_0202440C` -> `SaveData_TVBroadcast`
* `sub_02024420` -> `SaveData_PSSBoxes`
* `sub_0202442C` -> `SaveData_MysteryGift`
* `sub_02024440` -> `SaveData_PalParkTransfer`
* `sub_0202444C` -> `SaveData_HallOfFame`
* `sub_02024458` -> `SaveData_SaveHallOfFame`
* `sub_02024464` -> `SaveData_BattleRecording`
* `sub_0202447C` -> `SaveData_SaveBattleRecording`
* `sub_0202448C` -> `SaveData_BattleFrontierStage`
* `sub_020244A0` -> `SaveData_SaveBattleFrontierStage`

## Globals

* `Unk_020E5894` -> `gSaveTable`
* `Unk_020E5830` -> `gSaveTableSize`
* `Unk_020E5834` -> `gExtraSaveTable`
* `Unk_020E582C` -> `gExtraSaveTableSize`

### CODE REVIEW

File: include/savedata/save_table.h [hunk 1]
```diff
@@ -10,6 +10,70 @@
...
+    SAVE_TABLE_ENTRY_RANKINGS,
+    SAVE_TABLE_ENTRY_WIFI_LIST,
+    SAVE_TABLE_ENTRY_WIFI_HISTORY,
+    SAVE_TABLE_ENTRY_MYSTERY_GIFT,
+    SAVE_TABLE_ENTRY_PAL_PARK_TRANSFER,
+    SAVE_TABLE_ENTRY_CONTESTS,
+    SAVE_TABLE_ENTRY_SENTENCE,
+    SAVE_TABLE_ENTRY_EMAIL,
+    SAVE_TABLE_ENTRY_WIFI_QUESTIONS,
+    SAVE_TABLE_ENTRY_PSS_BOXES,
```
Comment (Nomura-RH) [inline include/savedata/save_table.h:71]: Noob question but what is this? Isn't PSS a Gen VI+ thing?

Comment (lhearachel) [reply include/savedata/save_table.h:71]: Pokemon Storage System:

https://bulbapedia.bulbagarden.net/wiki/Pok%C3%A9mon_Storage_System

Comment (Nomura-RH) [reply include/savedata/save_table.h:71]: > Pokemon Storage System:
> 
> https://bulbapedia.bulbagarden.net/wiki/Pok%C3%A9mon_Storage_System

O

What if we just called it `SAVE_TABLE_ENTRY_BOXES`?

Comment (lhearachel) [reply include/savedata/save_table.h:71]: I'm not against it, but I think the extra clarity here is good. If anything, maybe `PCBoxes`.

Comment (Nomura-RH) [reply include/savedata/save_table.h:71]: > I'm not against it, but I think the extra clarity here is good. If anything, maybe `PCBoxes`.

Sounds alright

**Pull request merged**